tests/test_servefile.py

```diff
@@ -20,6 +20,23 @@
     connrefused_exc = socket.error
 
 
+def _get_port_from_env(var_name, default):
+    if var_name not in os.environ:
+        return default
+    port = int(os.environ[var_name])
+    if port == 0:
+        # do a one-time port selection for a free port, use it for all tests
+        s = socket.socket()
+        s.bind(('', 0))
+        port = s.getsockname()[1]
+        s.close()
```

Comment:
I would personally extract that low-level code to a dedicated function "find_unused_port" or so to keep `_get_port_from_env` high-level and easy to understand.

Reply:
Hmm I get where you're going to, but it doesn't sit right with me. Having a three-line method and another three-line method feels not as good as having a single method. In my perception these utility functions feel like clutter in my tests and I'd like to have rather less of them than more.